docs/eventing/samples/writing-ra-source/01-theory.md

```diff
@@ -0,0 +1,94 @@
+
+[The specification](https://github.com/knative/eventing/blob/master/docs/spec/sources.md)
+for Knative Eventing Sources contains a number of requirements that
+together define a well-behaved Knative Source
+
+
+To achieve this, there are several separations of concerns that we have to keep in mind:
+1. A controller to run our Event Source and reconcile the underlying deployments
+2. A ‘receive adapter’ which imports the actual events
+3. A series of identifying characteristics for our event
+4. Transporting a valid event to the serverless system for further processing
+
+There are also two different classes of developer to consider:
+1. A "contributor" knows about the foreign protocol but not a Knative expert.
+2. Knative-eventing expert knows how knative eventing components are implemented, configured and deployed, but is not an expert in all the foreign protocols that sources may implement.
+These two roles will often not be the same person.  We want to confine the job of the "contributor" to implementing the `Receive Adapter`, and specifying what configuration their adapter needs connect, subscribe, or do whatever it does.
+
+The Knative-eventing developer exposes protocol configuration as part of the `CRD`, and the controller passes configuration (which may include resolved data like URLs) to the `Recieve Adapter`.
+
+API Resources required
+* `KubeClientSet.Appsv1.Deployment` (Inherited via eventing base reconciler)
+Used to deploy the Receive Adapter for ‘importing’ events
+* `EventingClientSet.EventingV1Alpha1` (Inherited via eventing base reconciler)
+Used to interact with Events within the knative system
+* `SourceClientSet.SourcesV1Alpha1`
+Used for source -- in this case, `samplesource` -- specific config and translated to the underlying deployment (via the inherited KubeClientSet)
+
+To ease writing a new event source, the eventing subsystem has offloaded several core functionalities (via injection) to the `eventing-sources-controller`.
+
+
+Fig 1. - Via shared [Knative Dependency Injection](https://docs.google.com/presentation/d/1aK5xCBv7wbfdDZAvnUE4vGWGk77EYZ6AbL0OR1vKPq8/edit#slide=id.g596dcbbefb_0_40)
+presentation in the knative community drive
+
+
+Specifically, the `clientset`, `cache`, `informers`, and `listers` can all be generated and shared. Thus, they can be generated, imported, and assigned to the underlying reconciler when creating a new controller source implementation:
+
+```golang
+import (
+…
+sampleSourceClient "knative.dev/sample-source/pkg/client/injection/client"
+samplesourceinformer “knative.dev/sample-source/pkg/client/injection/informers/samples/v1alpha1/samplesource"
+)
+…
+sampleSourceInformer := samplesourceinformer.Get(ctx)
+
+r := &Reconciler{
+...
+		samplesourceClientSet: sampleSourceClient.Get(ctx),
+		samplesourceLister:    sampleSourceInformer.Lister(),
+...
+```
+Ensure that the specific source subdirectory has been added to the injection portion of the `hack/update-codegen.sh` script.
```

Comment:
This needs clarification with a tiny code snippet, maybe in the form of a `diff`.